src/extension/exponent/exponentPlatform.ts

```diff
@@ -40,84 +40,86 @@ export class ExponentPlatform extends GeneralMobilePlatform {
-                })
-                .then(exponentUrl => {
-                    let exponentPage = vscode.window.createWebviewPanel("Expo QR Code", "Expo QR Code", vscode.ViewColumn.Two, { });
-                    exponentPage.webview.html = this.qrCodeContentProvider.provideTextDocumentContent(vscode.Uri.parse(exponentUrl));
-                    return exponentUrl;
-                })
-                .then(exponentUrl => {
-                    if (!exponentUrl) {
-                        return Q.reject<void>(ErrorHelper.getInternalError(InternalErrorCode.ExpectedExponentTunnelPath));
-                    }
-                    this.exponentTunnelPath = exponentUrl;
-                    const outputMessage = localize("ApplicationIsRunningOnExponentOpenToSeeIt", "Application is running on Expo. Open your Expo app at {0} to see it.", this.exponentTunnelPath);
-                    this.logger.info(outputMessage);
+                                return XDL.getUrl(this.projectPath, { dev: true, minify: false, hostType: "localhost" });
+                            case "tunnel":
+                            default:
+                                return XDL.getUrl(this.projectPath, { dev: true, minify: false });
+                        }
+                    })
+                    .then(exponentUrl => {
+                        return "exp://" + url.parse(exponentUrl).host;
+                    })
+                    .catch(reason => {
+                        return Promise.reject<string>(reason);
+                    })
+                    .then(exponentUrl => {
+                        let exponentPage = vscode.window.createWebviewPanel("Expo QR Code", "Expo QR Code", vscode.ViewColumn.Two, { });
+                        exponentPage.webview.html = this.qrCodeContentProvider.provideTextDocumentContent(vscode.Uri.parse(exponentUrl));
+                        return exponentUrl;
+                    })
+                    .then(exponentUrl => {
+                        if (!exponentUrl) {
+                            return Promise.reject<void>(ErrorHelper.getInternalError(InternalErrorCode.ExpectedExponentTunnelPath));
+                        }
+                        this.exponentTunnelPath = exponentUrl;
+                        const outputMessage = localize("ApplicationIsRunningOnExponentOpenToSeeIt", "Application is running on Expo. Open your Expo app at {0} to see it.", this.exponentTunnelPath);
+                        this.logger.info(outputMessage);
 
-                    return Q.resolve(void 0);
-                });
+                        return Promise.resolve();
+                    });
+            });
         });
     }
 
-    public loginToExponentOrSkip(expoHostType?: "tunnel" | "lan" | "local") {
+    public loginToExponentOrSkip(expoHostType?: "tunnel" | "lan" | "local"): Promise<any> {
         if (expoHostType !== "tunnel") {
-            return  Q({});
+            return  Promise.resolve();
```

Comment:
Does the `Promise.resolve()`expression return the same value as `Q({})`? Will it work without problems with this change?

Reply:
`Q({})` expression just promisfies empty object, as `Promise.resolve` returns resolved empty promise that we can use to chain further expressions.